socketIO_client/heartbeats.py

```diff
@@ -31,7 +31,8 @@ def run(self):
                     interval_in_seconds = self._hurry_interval_in_seconds
                 else:
                     interval_in_seconds = self._relax_interval_in_seconds
-                self._rest.wait(interval_in_seconds)
+                if self._rest and hasattr(self._rest, 'wait'):
```

Comment:
If this is not a valid situation, there should be an assert guard probably in line 20